spec/elastic/transport/opentelemetry_spec.rb

```diff
@@ -0,0 +1,179 @@
+        expect(described_class::ENDPOINT_PATH_REGEXPS['search'][0]).to eq(/\/_search$/)
+      end
+    end
+
+    context 'when a request is instrumented' do
+      let(:body) do
+        { query: { match: { password: { query: 'secret'} } } }
+      end
+
+      it 'creates a span and omits db.statement' do
+        client.perform_request('GET', '/_search', nil, body, nil, ["/_search", "/{index}/_search"], 'search')
+
+        expect(span.name).to eql('search')
+        expect(span.attributes['db.operation']).to eq('search')
+        expect(span.attributes['db.statement']).to be_nil
+        expect(span.attributes['http.request.method']).to eq('GET')
+        expect(span.attributes['server.address']).to eq('localhost')
+        expect(span.attributes['server.port']).to eq(9200)
+      end
+
+      context 'when body is sanitized' do
+        context 'no custom keys' do
+          let(:sanitized_body) do
+            { query: { match: { password: 'REDACTED' } } }
+          end
+
+          around(:example) do |ex|
+            body_strategy = ENV[described_class::ENV_VARIABLE_BODY_STRATEGY]
+            ENV[described_class::ENV_VARIABLE_BODY_STRATEGY]  = 'sanitize'
+            ex.run
+            ENV[described_class::ENV_VARIABLE_BODY_STRATEGY] = body_strategy
+          end
+
+          it 'sanitizes the body' do
+            client.perform_request('GET', '/_search', nil, body, nil, ["/_search", "/{index}/_search"], 'search')
+
+            expect(span.attributes['db.statement']).to eq(sanitized_body.to_json)
+          end
+        end
+
+        context 'with custom keys' do
+          let(:body) do
+            { query: { match: { sensitive: { query: 'secret'} } } }
+          end
+
+          let(:sanitized_body) do
+            { query: { match: { sensitive: 'REDACTED' } } }
+          end
+
+          around(:example) do |ex|
+            body_strategy = ENV[described_class::ENV_VARIABLE_BODY_STRATEGY]
+            ENV[described_class::ENV_VARIABLE_BODY_STRATEGY] = 'sanitize'
+
+            keys = ENV[described_class::ENV_VARIABLE_BODY_SANITIZE_KEYS]
+            ENV[described_class::ENV_VARIABLE_BODY_SANITIZE_KEYS] = 'sensitive'
+
+            ex.run
+
+            ENV[described_class::ENV_VARIABLE_BODY_STRATEGY] = body_strategy
+            ENV[described_class::ENV_VARIABLE_BODY_SANITIZE_KEYS] = keys
+          end
+
+          it 'sanitizes the body' do
+            client.perform_request('GET', '/_search', nil, body, nil, ["/_search", "/{index}/_search"], 'search')
+
+            expect(span.attributes['db.statement']).to eq(sanitized_body.to_json)
+          end
+        end
+      end
+
+      context 'a non-search endpoint' do
+        let(:body) do
+          { query: { match: { something: "test" } } }
+        end
+
+        it 'does not capture db.statement' do
+          client.perform_request(
+            'POST', 'foo/_delete_by_query', nil, body, nil, ["/{index}/_delete_by_query"], 'delete_by_query'
```

Comment:
@picandocodigo do you know of a request I could do in this spec that:
- has a body
- is a non-search endpoint
- doesn't require any setup (like index creation)

This tests fails because there is no `foo` index

Reply:
It looks like you can use `_all` instead of `foo`in this case. I tested this locally in a new cluster and it returns ok:
`client.perform_request('POST', '_all/_delete_by_query', {}, body, nil)`
The `perform_request` failed if I sent `nil` for parameters.